io/disk/Avago_storage_adapter/avago9361_vd.py

```diff
@@ -0,0 +1,199 @@
+            self.pdperarray = 2
+        elif self.raid_level == 'r50' or self.raid_level == 'r60':
+            self.pdperarray = 3
+        if self.raid_level == 'r0':
+            if 'cc' in str(self.name):
+                self.skip("CC not applicable for Raid0")
+
+    def test_createall(self):
+
+        """
+        Function to create different raid level
+        """
+        write_policy = ['WT', 'WB', 'AWB']
+        read_policy = ['nora', 'ra']
+        io_policy = ['direct', 'cached']
+        strip = [64, 128, 256, 512, 1024]
+        for write in write_policy:
+            for read in read_policy:
+                for iopolicy in io_policy:
+                    for stripe in strip:
+                        self.vd_create(write, read, iopolicy, stripe)
+                        self.vd_details()
+                        self.vd_delete()
+
+    def test_maxvd(self):
+
+        """
+        Function to create max VD
+        """
+        for _ in range(1, 17):
+            self.vd_create('WT', 'nora', 'direct', 1024)
+        self.vd_details()
+        self.vd_delete()
+        if self.raid_level == 'r0':
+            for disk in range(0, 3):
+                if disk > len(self.disk):
+                    self.raid_disk = self.disk[disk]
+                    for _ in range(1, 17):
+                        self.vd_create('WT', 'nora', 'direct', 512)
+                    self.vd_details()
+                self.vd_delete()
+
+    def test_cc(self):
+
+        """
+        Function to do consistency operations on VD
+        """
+        self.vd_create('WT', 'nora', 'direct', 256)
+        self.vd_details()
+        self.cc_operations()
+        self.vd_delete()
+
+    def vd_details(self):
+
+        """
+        Function to display the VD details
+        """
+        cmd = "./storcli64 /c%d/vall show" % self.controller
+        self.check_pass(cmd, "Failed to display VD configuration")
+
+    def vd_delete(self):
+
+        """
+        Function to delete the VD
+        """
+        cmd = "./storcli64 /c%d/vall delete force" % self.controller
+        self.check_pass(cmd, "Failed to delete VD")
+
+    def vd_create(self, write, read, iopolicy, stripe):
+
+        """
+        Function to create a VD
+        """
+        if self.raid_level == 'r0' or self.raid_level == 'r1' or \
+           self.raid_level == 'r5' or self.raid_level == 'r6':
+            cmd = "./storcli64 /c%d add vd %s size=%s drives=%s %s %s %s \
+                   strip=%d" % (self.controller, self.raid_level, self.size,
+                                self.raid_disk, write, read, iopolicy, stripe)
+            self.check_pass(cmd, "Failed to create raid")
+        else:
+            cmd = "./storcli64 /c%d add vd %s size=%s drives=%s PDperArray=%d %s %s %s \
+                   strip=%d" % (self.controller, self.raid_level, self.size,
+                                self.raid_disk, self.pdperarray, write, read,
+                                iopolicy, stripe)
```

Comment:
Since `self.pdperarray` is declared in if case, it would be better if this is also handled inside `if` instead of `else`